### ISSUE TITLE
toml_live update causes readfish to return 'ValueError: The flowcell cannot be split evenly'

### ISSUE THREAD

**Issue by TGSWilliams** [LooseLab/swordfish#7]
Hi,

I am trying to use swordfish balance for a Midnight COVID sequencing experiment.

Due to use of some samples with low Ct samples, some barcodes will fail to produce any reads which map to the COVID genome. Therefore it is not predictable before a run exactly how many samples will be run against the artic pipeline.

I have managed to set up minoTour with the artic pipeline, readfish and swordfish.  

However, when swordfish updates the original toml file to a .toml_live file which contains number of conditions which is not a factor of 32 (e.g. 7), readfish returns the following error: 'ValueError: The flowcell cannot be split evenly'. I can restart readfish with the same command, and again it will run until the first time swordfish updates the toml file.

Would it be possible to add dummy conditions in this instance, e.g. if only 7 conditions in toml_live file, to add an 8th dummy condition?

**Comment by mattloose** [comment]
Specifically the error you are seeing means that readfish is interpreting the toml file as asking to do different things on different regions of the flowcell and not different things to each barcode.

It shouldn't matter how many barcodes you have on the flowcell or how many of them will complete - minoTour will estimate the numbers that will complete based on the first hour or so of sequencing and will process them accordingly. ReadFish should reject hose amplicons which have sufficient coverage.

**Comment by TGSWilliams** [comment]
Hi,

I have tried readfish versions @guppy6, @guppy_6_minknow_5 @issue187, @issue 208 and also got the same error message. Is there a different guppy6 compatible version you would recommend?

Thanks

**Comment by mattloose** [comment]
HI,

name: readfish_swordfish 
channels:                                                        
  - bioconda                                                     
  - conda-forge                                                
  - defaults                                                       
dependencies:                                                
  - python=3.9                                                 
  - zlib                                                              
  - pip                                                               
  - pip:                                                              
    - read-until==3.0.0                                     
    - ont-pyguppy-client-lib==6.2.11              
    - git+https://github.com/LooseLab/readfish@issue187 
    - git+https://github.com/LooseLab/swordfish@main
    
 This is everything we usually install.

Also can I check if you are running the correct readfish command?

It should be: readfish barcode-targets

**Comment by TGSWilliams** [comment]
Hi,

Thanks for your help - I've now got swordfish up and running. I think the issue was using readfish targets rather than readfish barcode-targets as you suggest.

In case it helps anyone else, I also got held up by the fact that the barcoding kit name on minknow is SQK-RBK110-96 vs the product name of SQK-RBK110.96.

I will let you know how it impacts the runs once I have done a few with swordfish/readfish working on half of the channels in the flowcell.
